lib/cadet/jobs/autograder/plagiarism_checker.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Cadet.Autograder.PlagiarismChecker do
+  @moduledoc """
+  This module performs the plagiarism check script and sends results to
+  Amazon S3.
+  """
+  use Cadet, :context
+  use Que.Worker
+
+  import Ecto.Query
+
+  require Logger
+
+  alias Cadet.Assessments.Assessment
+
+  @bucket_name :cadet
+               |> Application.fetch_env!(:plagiarism_check_vars)
+               |> Keyword.get(:bucket_name)
+
+  @plagiarism_script_path :cadet
+                          |> Application.fetch_env!(:plagiarism_check_vars)
+                          |> Keyword.get(:plagiarism_script_path)
+
+  def perform(assessment_id) when is_ecto_id(assessment_id) do
+    Logger.info("Running plagiarism check on Assessment #{assessment_id}")
+
+    case System.cmd("python", [
+           @plagiarism_script_path,
+           "--assessment_id",
+           to_string(assessment_id)
+         ]) do
+      {result, 0} ->
+        result
+        |> Jason.decode!()
+        |> Map.get("assessment_id")
```

Comment:
I don't quite get why you're doing this. Won't the result from here equal to just calling `to_string(assessment_id)`?

Reply:
The real script itself does the plagiarism checking calls (database connection, writes answers to files, generates a report for said files through an online service). The output of the script is a `json` string that contains the `assessment_id`, but it has a side effect of producing a folder containing all the reports.

Reply:
Yea but my question is why do you need to do this? Isn't `result |> Jason.decode!() |> Map.get("assessment_id") == to_string(assessment_id)`? If so why not just use `to_string(assessment_id)`?

I mean proper python script should return non-zero exit code if there's an error, so the pattern matching wouldn't even go here in the first place if it failed right?